static/js/channel.js

```diff
@@ -31,11 +31,76 @@ Table.renderTable('table', consumerTableOpts, function (tr, item, all) {
   Table.renderCell(tr, 1, queueLink)
   Table.renderCell(tr, 2, ack, 'center')
   Table.renderCell(tr, 3, exclusive, 'center')
-  Table.renderCell(tr, 4, item.prefetch_count, 'right')
+  Table.renderCell(tr, 4, Helpers.formatNumber(item.prefetch_count), 'right')
 })
 
+const prefetchForm = (cb) => {
+  const input = document.createElement('input')
+  input.type = 'number'
+
+  const save = document.createElement('button')
+  save.type = 'submit'
+  save.title = 'Save'
+  save.classList.add('btn-icon', 'btn-submit')
+
+  const reset = document.createElement('button')
+  reset.type = 'reset'
+  reset.title = 'Reset'
+  reset.classList.add('btn-icon', 'btn-reset')
+
+  const form = document.createElement('form')
+  form.classList.add('prefetch-form')
+  form.addEventListener('submit', (event) => {
+    event.preventDefault()
+    const prefetch = parseInt(input.value)
+    HTTP.request('PUT', channelUrl, { body: { prefetch } })
+      .then((r) => {
+        if (!(r && r.is_error)) {
+          cb(prefetch)
+        }
```

Comment:
Show any alert if it fails? 

Can `r` be undefined?

Reply:
There is an automatic alert, triggering an popup, if we get back an error. Handled in http.js 

yes, `r` is undefined when no body is returned.